cypress/integration/A_000Login_spec.js

```diff
@@ -1,30 +1,82 @@
-/***
-Arrange - setup inital app state
-    - vist a web page
-    - query for an element 
-Act - take action 
-    - interact with that element
-Assert - make an assertion 
-    - Make assertion on web page
- ***/
+const token = "token";
+const userDetails = {
+    "email": "maggie@summit.com",
+    "first_name": "Maggie",
+    "id": 1,
+    "is_staff": true,
+    "last_name": "Huang",
+};
 
 describe("A:00Login_spec", () => {
-  it("Login Failed", () => {
-    cy.server();
-    cy.route({
-      method: "GET",
-      url: "/auth_token",
-      response: '{ "detail": "Method "GET" not allowed."}',
-      status: 400,
-    });
-  });
-
-  it("Login", () => {
-    cy.visit("http://localhost:3000/");
-    cy.get(".email").type("maggie@summit.com");
-    cy.get(".password").type("password");
-    cy.get('[type="checkbox"]').click();
-    cy.get('[type="submit"]').click();
-    cy.contains("Scheduler");
-  });
+    it("Loads login page", () => {
+        cy.visit("");
+        cy.get("[data-cy=signInButton");
+    });
+
+    it("Toggles Remember Me", () => {
+        cy.get("[data-cy=rememberMe]").click();
+        cy.get("[data-cy=rememberMe]").should("be.checked");
+    });
+
+    it("Disabled with only an email", () => {
+        cy.get("[data-cy=emailField]").fastType("maggie@summit.com");
+        cy.get("[data-cy=signInButton").should("be.disabled");
+        cy.get("[data-cy=emailField]").clear();
+    });
+
+    it("Disabled with only a password", () => {
+        cy.get("[data-cy=passwordField]").fastType("password");
+        cy.get("[data-cy=signInButton").should("be.disabled");
+    });
+
+    it("Enables submit with credentials", () => {
+        cy.get("[data-cy=emailField]").fastType("maggie@summit.com");
+        cy.get("[data-cy=signInButton").should("not.be.disabled");
+    });
+
+    it("Displays error on failed submit", () => {
+        cy.server();
+        cy.route({
+            "method": "POST",
+            "response": "Invalid credentials",
+            "status": 401,
+            "url": "/auth_token/",
+        });
+        cy.get("[data-cy=signInButton]").click();
+        cy.get("[data-cy=errorMessage]");
```

Comment:
Does this actually validate if the error message appears?

Reply:
Yep. The component isn't rendered if there is no error.